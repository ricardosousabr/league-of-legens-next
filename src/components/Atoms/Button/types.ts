import React, { ReactNode } from 'react';

 export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  href?: string;
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  as: string
}