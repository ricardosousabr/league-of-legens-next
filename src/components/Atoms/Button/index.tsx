import { ButtonProps } from './types'

export default function Button({children, type, onClick, href}:ButtonProps) {
  const buttonType = {
    type: type,
    onClick: onClick,
  }
  const anchorType = {
    as: "a",
    href: href,
  }

  return <button { ...href ? anchorType : buttonType }>{children}</button>
}