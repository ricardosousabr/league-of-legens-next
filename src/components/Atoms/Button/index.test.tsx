import renderer from "react-test-renderer";
import Button from "./index";

test("Link renders correctly", () => {
  const tree = renderer.create(<Button onClick={() => console.log("Hello")} type="button">Click</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
