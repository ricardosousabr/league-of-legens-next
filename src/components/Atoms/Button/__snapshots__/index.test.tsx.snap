// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Click
</button>
`;
